### PR TITLE
automatically generate linker version script for adapter libraries

### PR DESCRIPTION
**Pull request by pbalcer**

### CODE REVIEW

**Review by kbenzie — APPROVED** [review]
This makes a lot of sense.

**Review by omarahmed1111 — APPROVED** [review]
LGTM

File: scripts/templates/adapter.def.in.mako [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<%!
+import re
+from templates import helper as th
+%><%
+    n=namespace
+    N=n.upper()
+
+    x=tags['$x']
+    X=x.upper()
+%>
```
Comment (PatKamin) [inline scripts/templates/adapter.def.in.mako:10]: Add a `\` at the EOL here if empty lines at the beginnings of both scripts is not intended.

Comment (pbalcer) [reply scripts/templates/adapter.def.in.mako]: done

**Review by PatKamin — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**